Makefile

```diff
@@ -0,0 +1,108 @@
+NAP = 15
+SLEEP = 60
+DOCKER_COMPOSE_DEV = docker-compose-dev.yml
+DOCKER_TEST = docker exec -i anodot-agent pytest -x
+DOCKER_TEST_DEV = docker exec -i anodot-agent pytest -x -vv
+
+##------------
+## RELEASE
+##------------
+all: build-all sleep test-all
+
+##-------------
+## DEVELOPMENT
+##-------------
+all-dev: clean-docker-volumes build-all-dev sleep test-all-dev
+
+##-----------------------
+## TEST SEPARATE SOURCES
+##-----------------------
+test-directory: prepare-source test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_directory_http.py
+
+test-elastic: prepare-source build-elastic setup-elastic test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_elastic_http.py
+
+test-influx: prepare-source build-influx test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_influx_http.py
+
+test-kafka: prepare-source build-kafka setup-kafka test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_kafka_http.py
+
+test-mongo: prepare-source build-mongo test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_mongo_http.py
+
+test-mysql: prepare-source build-mysql test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_mysql_http.py
+
+test-postgres: prepare-source build-postgres test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_postgres_http.py
+
+test-tcp: prepare-source test-destination
+	$(DOCKER_TEST_DEV) tests/test_pipelines/test_tcp_http.py
+
+test-destination: prepare-source nap
+	$(DOCKER_TEST_DEV) tests/test_destination.py
+
+##--------------------
+## DEPENDENCY TARGETS
+##--------------------
+build-all: get-streamsets-stages
+	docker-compose up -d --build
```

Comment:
For a production build, we need to specify `GIT_SHA1` argument
`docker-compose build --build-arg GIT_SHA1=$(git describe --dirty --always) && docker-compose up -d`